libsolidity/ast/Types.cpp

```diff
@@ -347,6 +347,8 @@ TypePointer IntegerType::binaryOperatorResult(Token::Value _operator, TypePointe
 	// All integer types can be compared
 	if (Token::isCompareOp(_operator))
 		return commonType;
+	if (Token::isShiftOp(_operator) && !isAddress()) // && !_other->isAddress())
```

Comment:
As @pirapira suggested further below, this should not return the common type of the operands but rather the type of the left operand.